### DISCUSSION TITLE
Can I call the GUI in Headless Mode to produce and save the Mask and Overlay automatically?

### DISCUSSION THREAD

**Discussion by zdhernandez** [hkchengrex/MiVOS#26] [Q&A]
@hkchengrex 
I would like to pass the location of a point or a polygon ("scribble") without opening the GUI ("headless mode") and automatically save the output Mask and the Overlay. How can I do that ?

**Comment by hkchengrex** [comment]
There isn't a direct way to do it with the current codebase. You need to first convert those features into a mask (use fbrs for points and s2m for scribbles) and propagation from there.

**Reply by zdhernandez** [reply]
@hkchengrex where in the codebase is this located at ?  which sections do I need to explore to follow your steps ?
the point and scribbles from the user click(s) comes in as a mask too ?

**Reply by hkchengrex** [reply]
Every type of interaction will be converted into a mask. You can start from `interactive_gui.py`, `inference_core.py` (for propagation), `interact/s2m_controller.py` (for scribbles to mask), and `interact/fbrs_controller.py` (for clicks to mask)
